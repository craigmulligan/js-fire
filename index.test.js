const fire = require('./')
const util = require('util')
const fs = require('fs')
const execCb = require('child_process').exec
const exec = util.promisify(execCb)

describe('js-fire', () => {
  describe('function', () => {
    test('functions works with default as', async () => {
      const { stdout } = await exec('node examples/function.js')
      expect(stdout.trim()).toEqual('40')
    })
    test('helpText', async () => {
      const { stdout } = await exec('node examples/function.js --help')
      expect(stdout).toMatchSnapshot()
    })

    test('flag typo', async () => {
      try {
        await exec('node examples/function.js --numer=2')
      } catch (err) {
        expect(err.stdout.trim()).toMatchSnapshot()
      }
    })
  })

  describe('arrow function', () => {
    test('functions works unnamed arg', async () => {
      const { stdout } = await exec('node examples/arrow-function.js 20')
      expect(stdout.trim()).toEqual('40')
    })
    test('helpText', async () => {
      const { stdout } = await exec('node examples/arrow-function.js --help')
      expect(stdout).toMatchSnapshot()
    })
  })

  describe('circular reference', async () => {
    test('should compile', async () => {
      const { stdout } = await exec(
        'node examples/circular-reference.js half --number 20',
      )
      expect(stdout.trim()).toEqual('10')
    })
  })

  describe('lib', () => {
    // tests whether you can call a arbitratry library via the js-fire api.

    test('write a file with fs', async () => {
      const path = __dirname + '/test.txt'
      await exec(
        `node bin/index.js fs -- writeFileSync --path ${path} --data "hiii"`,
      )
      const constents = fs.readFileSync(path)
      expect(constents.toString()).toEqual('hiii')
      // cleanup
      fs.unlinkSync(path)
    })

    test('helpText', async () => {
      const { stdout } = await exec('node bin/index.js fs -h')
      expect(stdout.trim()).toMatchSnapshot()
    })
  })

  describe('cli', () => {
    const opts = { cwd: 'examples/cli' }

    beforeAll(async () => {
      await exec(`yarn link`, opts)
    })

    test('helpText', async () => {
      const { stdout } = await exec('js-fire-example -h')
      // Command usage should ref js-fire
      expect(stdout.trim()).toMatchSnapshot()
    })
  })

  describe('interactive', () => {
    const keys = {
      up: '\x1B\x5B\x41',
      down: '\x1B\x5B\x42',
      enter: '\x0D',
      space: '\x20',
    }

    test('object with subcommand', done => {
      const ps = execCb('node examples/object.js half -i', (err, stdout) => {
        expect(stdout.trim()).toContain('10')
        done()
      })

      ps.stdin.write(`20${keys.enter}`)
    })

    test('object from root', done => {
      const ps = execCb('node examples/object.js misc -i', (err, stdout) => {
        console.log(stdout)
        expect(stdout.trim()).toContain('greetings hobo!')
        done()
      })

      setTimeout(() => ps.stdin.write(`info\r\n`), 100)
      setTimeout(() => ps.stdin.write(`greeting\r\n`), 200)
      setTimeout(() => ps.stdin.write(`hobo\r\n`), 400)
    })

    test('object from root', done => {
      const ps = execCb('node examples/object.js -i', (err, stdout) => {
        console.log(stdout)
        expect(stdout.trim()).toContain('greetings hobo!')
        done()
      })

      setTimeout(() => ps.stdin.write(`misc\r\n`), 100)
      setTimeout(() => ps.stdin.write(`info\r\n`), 200)
      setTimeout(() => ps.stdin.write(`greeting\r\n`), 400)
      setTimeout(() => ps.stdin.write(`hobo\r\n`), 600)
    })
  })

  describe('bin', () => {
    beforeAll(async () => {
      await exec(`yarn link`)
    })

    test('helpText', async () => {
      const { stdout } = await exec('js-fire -h')
      // Command usage should ref js-fire
      expect(stdout.trim()).toMatchSnapshot()
    })

    test('helpText with subcommand without help flag', async () => {
      const { stdout } = await exec('js-fire fs -h')
      // Command usage should ref js-fire
      expect(stdout.trim()).toMatchSnapshot()
    })

    test('helpText with subcommand with help flag', async () => {
      const { stdout } = await exec('js-fire fs -h')
      expect(stdout.trim()).toMatchSnapshot()
    })

    test('helpText with incorrect subcommand name', async () => {
      // this should return a suggested command in the output
      try {
        const { stdout } = await exec('js-fire fs appendF')
        // this should return an exit code 1
      } catch (err) {
        expect(err.stdout.trim()).toMatchSnapshot()
      }
    })

    test('bin with named args', async () => {
      const { stdout } = await exec(
        `js-fire fs -- existsSync --path ${__filename}`,
      )
      expect(stdout.trim()).toEqual('true')
    })

    test('bin with positional args', async () => {
      const { stdout } = await exec(`js-fire fs existsSync ${__filename}`)
      expect(stdout.trim()).toEqual('true')
    })
  })

  describe('object', () => {
    test('half', async () => {
      const { stdout } = await exec('node examples/object.js half --number 20')
      expect(stdout.trim()).toEqual('10')
    })

    test('double', async () => {
      const { stdout } = await exec(
        'node examples/object.js double --number 20',
      )
      expect(stdout.trim()).toEqual('40')
    })

    test('works without named args', async () => {
      const { stdout } = await exec('node examples/object.js double 20')
      expect(stdout.trim()).toEqual('40')
    })

    test('works with deeply nested objects', async () => {
      const { stdout } = await exec(
        'node examples/object.js misc hello --name hobochild',
      )
      expect(stdout.trim()).toEqual('hi hobochild')
    })

    test('helpText', async () => {
      const { stdout } = await exec('node examples/object.js --help')
      expect(stdout).toMatchSnapshot()
    })

    test('helpText with subcommand', async () => {
      const { stdout } = await exec('node examples/object.js double --help')
      expect(stdout).toMatchSnapshot()
    })

    test('command not found', async () => {
      // test case for:
      // https://github.com/hobochild/js-fire/issues/22
      const { stdout } = await exec(
        'node examples/object.js misc; echo "exit code $?"',
      )
      expect(stdout).toMatchSnapshot()
    })

    test('deeply nested', async () => {
      const { stdout } = await exec(
        'node examples/object.js misc info greeting --name hobochild',
      )
      expect(stdout.trim()).toEqual('greetings hobochild!')
    })

    test('deeply nested with typo', async () => {
      const { stdout } = await exec(
        'node examples/object.js misc info greeting --name hobochild',
      )
      expect(stdout).toMatchSnapshot()
    })
  })

  test('async functions', async () => {
    const { stdout } = await exec('node examples/async.js --number 20')
    expect(stdout.trim()).toEqual('40')
  })

  describe('multi-args', () => {
    test('multi unname args', async () => {
      const { stdout } = await exec('node examples/multi-args.js 30 20')
      expect(stdout.trim()).toEqual('10')
    })
  })

  test('multi named args', async () => {
    const { stdout } = await exec('node examples/multi-args.js --b 30 --a 20')
    expect(stdout.trim()).toEqual('-10')
  })

  test('works with default (eval) values', async () => {
    const { stdout } = await exec('node examples/multi-args.js --b 1')
    const answer = Math.PI - 1
    expect(stdout.trim()).toEqual(answer.toString())
  })

  test('helpText', async () => {
    const { stdout } = await exec('node examples/multi-args.js --help')
    expect(stdout.trim()).toMatchSnapshot()
  })
})
