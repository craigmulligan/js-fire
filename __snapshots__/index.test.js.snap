// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`js-fire arrow function helpText 1`] = `
"USAGE:
	node arrow-function.js --number=<number>
"
`;

exports[`js-fire bin helpText 1`] = `
"USAGE:
	js-fire --modulePath=<modulePath>"
`;

exports[`js-fire bin helpText with incorrect subcommand name 1`] = `
"Error: Command appendF not found
Did you mean: appendFile ?

USAGE:
	js-fire fs appendF

COMMANDS:

	Dirent
	FileReadStream --path=<path>  --options=<options> 
	FileWriteStream --path=<path>  --options=<options> 
	ReadStream --path=<path>  --options=<options> 
	Stats
	WriteStream --path=<path>  --options=<options> 
	access --path=<path>  --mode=<mode>  --callback=<callback> 
	accessSync --path=<path>  --mode=<mode> 
	appendFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	appendFileSync --path=<path>  --data=<data>  --options=<options> 
	chmod --path=<path>  --mode=<mode>  --callback=<callback> 
	chmodSync --path=<path>  --mode=<mode> 
	chown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	chownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	close --fd=<fd>  --callback=<callback> 
	closeSync --fd=<fd> 
	copyFile --src=<src>  --dest=<dest>  --flags=<flags>  --callback=<callback> 
	copyFileSync --src=<src>  --dest=<dest>  --flags=<flags> 
	createReadStream --path=<path>  --options=<options> 
	createWriteStream --path=<path>  --options=<options> 
	exists --path=<path>  --callback=<callback> 
	existsSync --path=<path> 
	fchmod --fd=<fd>  --mode=<mode>  --callback=<callback> 
	fchmodSync --fd=<fd>  --mode=<mode> 
	fchown --fd=<fd>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	fchownSync --fd=<fd>  --uid=<uid>  --gid=<gid> 
	fdatasync --fd=<fd>  --callback=<callback> 
	fdatasyncSync --fd=<fd> 
	fstat --fd=<fd>  --options=<options>  --callback=<callback> 
	fstatSync --fd=<fd>  --options=undefined 
	fsync --fd=<fd>  --callback=<callback> 
	fsyncSync --fd=<fd> 
	ftruncate --fd=<fd>  --len=0  --callback=<callback> 
	ftruncateSync --fd=<fd>  --len=0 
	futimes --fd=<fd>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	futimesSync --fd=<fd>  --atime=<atime>  --mtime=<mtime> 
	lchown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	lchownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	link --existingPath=<existingPath>  --newPath=<newPath>  --callback=<callback> 
	linkSync --existingPath=<existingPath>  --newPath=<newPath> 
	lstat --path=<path>  --options=<options>  --callback=<callback> 
	lstatSync --path=<path>  --options=undefined 
	mkdir --path=<path>  --options=<options>  --callback=<callback> 
	mkdirSync --path=<path>  --options=<options> 
	mkdtemp --prefix=<prefix>  --options=<options>  --callback=<callback> 
	mkdtempSync --prefix=<prefix>  --options=<options> 
	open --path=<path>  --flags=<flags>  --mode=<mode>  --callback=<callback> 
	openSync --path=<path>  --flags=<flags>  --mode=<mode> 
	read --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	readFile --path=<path>  --options=<options>  --callback=<callback> 
	readFileSync --path=<path>  --options=<options> 
	readSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position> 
	readdir --path=<path>  --options=<options>  --callback=<callback> 
	readdirSync --path=<path>  --options=<options> 
	readlink --path=<path>  --options=<options>  --callback=<callback> 
	readlinkSync --path=<path>  --options=<options> 
	realpath --p=<p>  --options=<options>  --callback=<callback> 
	realpathSync --p=<p>  --options=<options> 
	rename --oldPath=<oldPath>  --newPath=<newPath>  --callback=<callback> 
	renameSync --oldPath=<oldPath>  --newPath=<newPath> 
	rmdir --path=<path>  --callback=<callback> 
	rmdirSync --path=<path> 
	stat --path=<path>  --options=<options>  --callback=<callback> 
	statSync --path=<path>  --options=undefined 
	symlink --target=<target>  --path=<path>  --type_=<type_>  --callback_=<callback_> 
	symlinkSync --target=<target>  --path=<path>  --type=<type> 
	truncate --path=<path>  --len=<len>  --callback=<callback> 
	truncateSync --path=<path>  --len=<len> 
	unlink --path=<path>  --callback=<callback> 
	unlinkSync --path=<path> 
	unwatchFile --filename=<filename>  --listener=<listener> 
	utimes --path=<path>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	utimesSync --path=<path>  --atime=<atime>  --mtime=<mtime> 
	watch --filename=<filename>  --options=<options>  --listener=<listener> 
	watchFile --filename=<filename>  --options=<options>  --listener=<listener> 
	write --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	writeFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	writeFileSync --path=<path>  --data=<data>  --options=<options> 
	writeSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>"
`;

exports[`js-fire bin helpText with subcommand with help flag 1`] = `
"Error: Command fs not found


USAGE:
	js-fire fs

COMMANDS:

	Dirent
	FileReadStream --path=<path>  --options=<options> 
	FileWriteStream --path=<path>  --options=<options> 
	ReadStream --path=<path>  --options=<options> 
	Stats
	WriteStream --path=<path>  --options=<options> 
	access --path=<path>  --mode=<mode>  --callback=<callback> 
	accessSync --path=<path>  --mode=<mode> 
	appendFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	appendFileSync --path=<path>  --data=<data>  --options=<options> 
	chmod --path=<path>  --mode=<mode>  --callback=<callback> 
	chmodSync --path=<path>  --mode=<mode> 
	chown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	chownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	close --fd=<fd>  --callback=<callback> 
	closeSync --fd=<fd> 
	copyFile --src=<src>  --dest=<dest>  --flags=<flags>  --callback=<callback> 
	copyFileSync --src=<src>  --dest=<dest>  --flags=<flags> 
	createReadStream --path=<path>  --options=<options> 
	createWriteStream --path=<path>  --options=<options> 
	exists --path=<path>  --callback=<callback> 
	existsSync --path=<path> 
	fchmod --fd=<fd>  --mode=<mode>  --callback=<callback> 
	fchmodSync --fd=<fd>  --mode=<mode> 
	fchown --fd=<fd>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	fchownSync --fd=<fd>  --uid=<uid>  --gid=<gid> 
	fdatasync --fd=<fd>  --callback=<callback> 
	fdatasyncSync --fd=<fd> 
	fstat --fd=<fd>  --options=<options>  --callback=<callback> 
	fstatSync --fd=<fd>  --options=undefined 
	fsync --fd=<fd>  --callback=<callback> 
	fsyncSync --fd=<fd> 
	ftruncate --fd=<fd>  --len=0  --callback=<callback> 
	ftruncateSync --fd=<fd>  --len=0 
	futimes --fd=<fd>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	futimesSync --fd=<fd>  --atime=<atime>  --mtime=<mtime> 
	lchown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	lchownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	link --existingPath=<existingPath>  --newPath=<newPath>  --callback=<callback> 
	linkSync --existingPath=<existingPath>  --newPath=<newPath> 
	lstat --path=<path>  --options=<options>  --callback=<callback> 
	lstatSync --path=<path>  --options=undefined 
	mkdir --path=<path>  --options=<options>  --callback=<callback> 
	mkdirSync --path=<path>  --options=<options> 
	mkdtemp --prefix=<prefix>  --options=<options>  --callback=<callback> 
	mkdtempSync --prefix=<prefix>  --options=<options> 
	open --path=<path>  --flags=<flags>  --mode=<mode>  --callback=<callback> 
	openSync --path=<path>  --flags=<flags>  --mode=<mode> 
	read --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	readFile --path=<path>  --options=<options>  --callback=<callback> 
	readFileSync --path=<path>  --options=<options> 
	readSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position> 
	readdir --path=<path>  --options=<options>  --callback=<callback> 
	readdirSync --path=<path>  --options=<options> 
	readlink --path=<path>  --options=<options>  --callback=<callback> 
	readlinkSync --path=<path>  --options=<options> 
	realpath --p=<p>  --options=<options>  --callback=<callback> 
	realpathSync --p=<p>  --options=<options> 
	rename --oldPath=<oldPath>  --newPath=<newPath>  --callback=<callback> 
	renameSync --oldPath=<oldPath>  --newPath=<newPath> 
	rmdir --path=<path>  --callback=<callback> 
	rmdirSync --path=<path> 
	stat --path=<path>  --options=<options>  --callback=<callback> 
	statSync --path=<path>  --options=undefined 
	symlink --target=<target>  --path=<path>  --type_=<type_>  --callback_=<callback_> 
	symlinkSync --target=<target>  --path=<path>  --type=<type> 
	truncate --path=<path>  --len=<len>  --callback=<callback> 
	truncateSync --path=<path>  --len=<len> 
	unlink --path=<path>  --callback=<callback> 
	unlinkSync --path=<path> 
	unwatchFile --filename=<filename>  --listener=<listener> 
	utimes --path=<path>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	utimesSync --path=<path>  --atime=<atime>  --mtime=<mtime> 
	watch --filename=<filename>  --options=<options>  --listener=<listener> 
	watchFile --filename=<filename>  --options=<options>  --listener=<listener> 
	write --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	writeFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	writeFileSync --path=<path>  --data=<data>  --options=<options> 
	writeSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>"
`;

exports[`js-fire bin helpText with subcommand without help flag 1`] = `
"Error: Command fs not found


USAGE:
	js-fire fs

COMMANDS:

	Dirent
	FileReadStream --path=<path>  --options=<options> 
	FileWriteStream --path=<path>  --options=<options> 
	ReadStream --path=<path>  --options=<options> 
	Stats
	WriteStream --path=<path>  --options=<options> 
	access --path=<path>  --mode=<mode>  --callback=<callback> 
	accessSync --path=<path>  --mode=<mode> 
	appendFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	appendFileSync --path=<path>  --data=<data>  --options=<options> 
	chmod --path=<path>  --mode=<mode>  --callback=<callback> 
	chmodSync --path=<path>  --mode=<mode> 
	chown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	chownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	close --fd=<fd>  --callback=<callback> 
	closeSync --fd=<fd> 
	copyFile --src=<src>  --dest=<dest>  --flags=<flags>  --callback=<callback> 
	copyFileSync --src=<src>  --dest=<dest>  --flags=<flags> 
	createReadStream --path=<path>  --options=<options> 
	createWriteStream --path=<path>  --options=<options> 
	exists --path=<path>  --callback=<callback> 
	existsSync --path=<path> 
	fchmod --fd=<fd>  --mode=<mode>  --callback=<callback> 
	fchmodSync --fd=<fd>  --mode=<mode> 
	fchown --fd=<fd>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	fchownSync --fd=<fd>  --uid=<uid>  --gid=<gid> 
	fdatasync --fd=<fd>  --callback=<callback> 
	fdatasyncSync --fd=<fd> 
	fstat --fd=<fd>  --options=<options>  --callback=<callback> 
	fstatSync --fd=<fd>  --options=undefined 
	fsync --fd=<fd>  --callback=<callback> 
	fsyncSync --fd=<fd> 
	ftruncate --fd=<fd>  --len=0  --callback=<callback> 
	ftruncateSync --fd=<fd>  --len=0 
	futimes --fd=<fd>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	futimesSync --fd=<fd>  --atime=<atime>  --mtime=<mtime> 
	lchown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	lchownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	link --existingPath=<existingPath>  --newPath=<newPath>  --callback=<callback> 
	linkSync --existingPath=<existingPath>  --newPath=<newPath> 
	lstat --path=<path>  --options=<options>  --callback=<callback> 
	lstatSync --path=<path>  --options=undefined 
	mkdir --path=<path>  --options=<options>  --callback=<callback> 
	mkdirSync --path=<path>  --options=<options> 
	mkdtemp --prefix=<prefix>  --options=<options>  --callback=<callback> 
	mkdtempSync --prefix=<prefix>  --options=<options> 
	open --path=<path>  --flags=<flags>  --mode=<mode>  --callback=<callback> 
	openSync --path=<path>  --flags=<flags>  --mode=<mode> 
	read --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	readFile --path=<path>  --options=<options>  --callback=<callback> 
	readFileSync --path=<path>  --options=<options> 
	readSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position> 
	readdir --path=<path>  --options=<options>  --callback=<callback> 
	readdirSync --path=<path>  --options=<options> 
	readlink --path=<path>  --options=<options>  --callback=<callback> 
	readlinkSync --path=<path>  --options=<options> 
	realpath --p=<p>  --options=<options>  --callback=<callback> 
	realpathSync --p=<p>  --options=<options> 
	rename --oldPath=<oldPath>  --newPath=<newPath>  --callback=<callback> 
	renameSync --oldPath=<oldPath>  --newPath=<newPath> 
	rmdir --path=<path>  --callback=<callback> 
	rmdirSync --path=<path> 
	stat --path=<path>  --options=<options>  --callback=<callback> 
	statSync --path=<path>  --options=undefined 
	symlink --target=<target>  --path=<path>  --type_=<type_>  --callback_=<callback_> 
	symlinkSync --target=<target>  --path=<path>  --type=<type> 
	truncate --path=<path>  --len=<len>  --callback=<callback> 
	truncateSync --path=<path>  --len=<len> 
	unlink --path=<path>  --callback=<callback> 
	unlinkSync --path=<path> 
	unwatchFile --filename=<filename>  --listener=<listener> 
	utimes --path=<path>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	utimesSync --path=<path>  --atime=<atime>  --mtime=<mtime> 
	watch --filename=<filename>  --options=<options>  --listener=<listener> 
	watchFile --filename=<filename>  --options=<options>  --listener=<listener> 
	write --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	writeFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	writeFileSync --path=<path>  --data=<data>  --options=<options> 
	writeSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>"
`;

exports[`js-fire cli helpText 1`] = `
"Error: Command not found

USAGE:
	js-fire-example

DESCRIPTION: 
	I am a math machine


COMMANDS:

	add --n1=3.141592653589793  --n2=<n2> 
	double --number=<number> 
	half --number=<number> 

	misc
		brand
		hello --name=<name> 
		year"
`;

exports[`js-fire function flag typo 1`] = `
"Error: Flag numer not found
Did you mean: number ?
USAGE:
	node function.js --number=20"
`;

exports[`js-fire function helpText 1`] = `
"USAGE:
	node function.js --number=20
"
`;

exports[`js-fire helpText 1`] = `
"USAGE:
	node multi-args.js --a=3.141592653589793  --b=20  --noop=false  --randomString=hiii"
`;

exports[`js-fire lib helpText 1`] = `
"Error: Command fs not found


USAGE:
	node index.js fs

COMMANDS:

	Dirent
	FileReadStream --path=<path>  --options=<options> 
	FileWriteStream --path=<path>  --options=<options> 
	ReadStream --path=<path>  --options=<options> 
	Stats
	WriteStream --path=<path>  --options=<options> 
	access --path=<path>  --mode=<mode>  --callback=<callback> 
	accessSync --path=<path>  --mode=<mode> 
	appendFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	appendFileSync --path=<path>  --data=<data>  --options=<options> 
	chmod --path=<path>  --mode=<mode>  --callback=<callback> 
	chmodSync --path=<path>  --mode=<mode> 
	chown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	chownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	close --fd=<fd>  --callback=<callback> 
	closeSync --fd=<fd> 
	copyFile --src=<src>  --dest=<dest>  --flags=<flags>  --callback=<callback> 
	copyFileSync --src=<src>  --dest=<dest>  --flags=<flags> 
	createReadStream --path=<path>  --options=<options> 
	createWriteStream --path=<path>  --options=<options> 
	exists --path=<path>  --callback=<callback> 
	existsSync --path=<path> 
	fchmod --fd=<fd>  --mode=<mode>  --callback=<callback> 
	fchmodSync --fd=<fd>  --mode=<mode> 
	fchown --fd=<fd>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	fchownSync --fd=<fd>  --uid=<uid>  --gid=<gid> 
	fdatasync --fd=<fd>  --callback=<callback> 
	fdatasyncSync --fd=<fd> 
	fstat --fd=<fd>  --options=<options>  --callback=<callback> 
	fstatSync --fd=<fd>  --options=undefined 
	fsync --fd=<fd>  --callback=<callback> 
	fsyncSync --fd=<fd> 
	ftruncate --fd=<fd>  --len=0  --callback=<callback> 
	ftruncateSync --fd=<fd>  --len=0 
	futimes --fd=<fd>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	futimesSync --fd=<fd>  --atime=<atime>  --mtime=<mtime> 
	lchown --path=<path>  --uid=<uid>  --gid=<gid>  --callback=<callback> 
	lchownSync --path=<path>  --uid=<uid>  --gid=<gid> 
	link --existingPath=<existingPath>  --newPath=<newPath>  --callback=<callback> 
	linkSync --existingPath=<existingPath>  --newPath=<newPath> 
	lstat --path=<path>  --options=<options>  --callback=<callback> 
	lstatSync --path=<path>  --options=undefined 
	mkdir --path=<path>  --options=<options>  --callback=<callback> 
	mkdirSync --path=<path>  --options=<options> 
	mkdtemp --prefix=<prefix>  --options=<options>  --callback=<callback> 
	mkdtempSync --prefix=<prefix>  --options=<options> 
	open --path=<path>  --flags=<flags>  --mode=<mode>  --callback=<callback> 
	openSync --path=<path>  --flags=<flags>  --mode=<mode> 
	read --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	readFile --path=<path>  --options=<options>  --callback=<callback> 
	readFileSync --path=<path>  --options=<options> 
	readSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position> 
	readdir --path=<path>  --options=<options>  --callback=<callback> 
	readdirSync --path=<path>  --options=<options> 
	readlink --path=<path>  --options=<options>  --callback=<callback> 
	readlinkSync --path=<path>  --options=<options> 
	realpath --p=<p>  --options=<options>  --callback=<callback> 
	realpathSync --p=<p>  --options=<options> 
	rename --oldPath=<oldPath>  --newPath=<newPath>  --callback=<callback> 
	renameSync --oldPath=<oldPath>  --newPath=<newPath> 
	rmdir --path=<path>  --callback=<callback> 
	rmdirSync --path=<path> 
	stat --path=<path>  --options=<options>  --callback=<callback> 
	statSync --path=<path>  --options=undefined 
	symlink --target=<target>  --path=<path>  --type_=<type_>  --callback_=<callback_> 
	symlinkSync --target=<target>  --path=<path>  --type=<type> 
	truncate --path=<path>  --len=<len>  --callback=<callback> 
	truncateSync --path=<path>  --len=<len> 
	unlink --path=<path>  --callback=<callback> 
	unlinkSync --path=<path> 
	unwatchFile --filename=<filename>  --listener=<listener> 
	utimes --path=<path>  --atime=<atime>  --mtime=<mtime>  --callback=<callback> 
	utimesSync --path=<path>  --atime=<atime>  --mtime=<mtime> 
	watch --filename=<filename>  --options=<options>  --listener=<listener> 
	watchFile --filename=<filename>  --options=<options>  --listener=<listener> 
	write --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>  --callback=<callback> 
	writeFile --path=<path>  --data=<data>  --options=<options>  --callback=<callback> 
	writeFileSync --path=<path>  --data=<data>  --options=<options> 
	writeSync --fd=<fd>  --buffer=<buffer>  --offset=<offset>  --length=<length>  --position=<position>"
`;

exports[`js-fire object command not found 1`] = `
"Error: Command misc not found


USAGE:
	node object.js misc

COMMANDS:

	brand
	hello --name=<name> 

	info
		greeting --name=hobochild 

	year

exit code 1
"
`;

exports[`js-fire object deeply nested with typo 1`] = `
"greetings hobochild!
"
`;

exports[`js-fire object helpText 1`] = `
"Error: Command not found

USAGE:
	node object.js

DESCRIPTION: 
	I am a math machine


COMMANDS:

	add --n1=3.141592653589793  --n2=<n2> 
	double --number=<number> 
	half --number=<number> 

	misc
		brand
		hello --name=<name> 

		info
			greeting --name=hobochild 

		year
"
`;

exports[`js-fire object helpText with subcommand 1`] = `
"USAGE:
	node object.js double --number=<number>
"
`;
